### ISSUE TITLE
vim colorschemes not applied for sudo

### ISSUE THREAD

**Issue by waynongithub** [kovidgoyal/kitty#5574]
Thank you for Kitty!

**Describe the bug**
I'm using Linux Mint 20.2 MATE 
Kitty was installed with `curl -L https://sw.kovidgoyal.net/kitty/installer.sh | sh /dev/stdin`
vim 8.1 was installed with the Mint software manager. 

When I open a document with `sudo vim` in _mate-terminal,_ _gnome-terminal_ or _terminator_, the document is rendered in full color with the same colorscheme as without `sudo`. But in Kitty, `sudo vim` renders documents in black and white. I can select different colorschemes, but they all turn out black and white.
I'm sure this is not a bug, but I can't figure out which settings would fix this. 

**To Reproduce**
1. start vim with `vim <file>`: document is rendered in full color according to the colorscheme.
2. start vim with `sudo vim <file>`: document is rendered in black and white.

**Environment details**
```
kitty 0.26.2 created by Kovid Goyal
Linux ASUS 5.4.0-126-generic #142-Ubuntu SMP Fri Aug 26 12:12:57 UTC 2022 x86_64
Linux Mint 20.2 Uma ASUS /dev/tty

DISTRIB_ID=LinuxMint
DISTRIB_RELEASE=20.2
DISTRIB_CODENAME=uma
DISTRIB_DESCRIPTION="Linux Mint 20.2 Uma"
Running under: X11
Frozen: True
Paths:
  kitty: /media/bconf/settings/kitty.app/bin/kitty
  base dir: /media/bconf/settings/kitty.app/lib/kitty
  extensions dir: /media/bconf/settings/kitty.app/lib/kitty-extensions
  system shell: /bin/bash
Loaded config files:
  /home/me/.config/kitty/kitty.conf

Config options different from defaults:
bold_font             JetBrains Mono Bold
bold_italic_font      JetBrains Mono Bold Italic
disable_ligatures     2
enable_audio_bell     False
enabled_layouts       ['splits:split_axis=horizontal']
font_family           JetBrains Mono Medium
font_size             10.0
initial_window_height (960, 'px')
initial_window_width  (840, 'px')
italic_font           JetBrains Mono Italic
macos_thicken_font    0.3
modify_font:
    cell_height 100%
tab_bar_edge          1
tab_bar_style         powerline
tab_title_template    {index}:{title}
window_padding_width  FloatEdges(left=2.0, top=2.0, right=2.0, bottom=2.0)
Added shortcuts:
	alt+down →  neighboring_window down
	alt+left →  neighboring_window left
	alt+right →  neighboring_window right
	alt+up →  neighboring_window up
	ctrl+alt+enter →  launch --cwd=current
	f4 →  launch --location=split
	f7 →  layout_action rotate
	shift+down →  move_window down
	shift+left →  move_window left
	shift+right →  move_window right
	shift+up →  move_window up
Changed shortcuts:
	kitty_mod+e →  launch --location=vsplit --cwd=current
	kitty_mod+o →  launch --location=hsplit --cwd=current
Colors:
	color0                #2a2e38   
	color1                #d08785   
	color10               #9ec400   
	color11               #e7c547   
	color12               #7aa6da   
	color13               #b77ee0   
	color14               #54ced6   
	color2                #6f98b3   
	color3                #fdf8ce   
	color4                #a6b8cc   
	color5                #fcdbd9   
	color6                #ffd17f   
	color7                #f0fdff   
	color8                #71798a   
	color9                #ff3334   
	foreground            #f0fdff   

Important environment variables seen by the kitty process:
	PATH                                /media/bconf/settings/kitty.app/bin:/home/me/bin:/home/me/.local/bin:/home/me/.npm-global:/home/me/.npm-global/bin:/home/me/zz_programs/node-v16.13.2-linux-x64/bin:/home/me/zz_programs:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
	LANG                                en_US.UTF-8
	EDITOR                              vim
	SHELL                               /bin/bash
	DISPLAY                             :0
	USER                                me
	XDG_CONFIG_DIRS                     /etc/xdg/xdg-mate:/etc/xdg
	XDG_SESSION_PATH                    /org/freedesktop/DisplayManager/Session0
	LC_ADDRESS                          th_TH.UTF-8
	LC_NAME                             th_TH.UTF-8
	LC_MONETARY                         th_TH.UTF-8
	XDG_SEAT                            seat0
	XDG_SESSION_DESKTOP                 mate
	XDG_SESSION_TYPE                    x11
	XDG_GREETER_DATA_DIR                /var/lib/lightdm-data/me
	LC_PAPER                            th_TH.UTF-8
	XDG_CURRENT_DESKTOP                 MATE
	XDG_SEAT_PATH                       /org/freedesktop/DisplayManager/Seat0
	XDG_SESSION_CLASS                   user
	LC_IDENTIFICATION                   th_TH.UTF-8
	LC_TELEPHONE                        th_TH.UTF-8
	LC_MEASUREMENT                      th_TH.UTF-8
	XDG_VTNR                            7
	XDG_SESSION_ID                      c2
	XDG_RUNTIME_DIR                     /run/user/1000
	XDG_DATA_DIRS                       /usr/share/mate:/home/me/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share
	LC_NUMERIC                          th_TH.UTF-8

```

**Additional context**
TERM=xterm-kitty
I get the same result when I start kitty with `kitty --config NONE`  
I also tried after deleting .vimrc and .vim, same result.
I also tried with `sudo -E vim <file>`, still black and white.

**Comment by kovidgoyal** [comment]
It will be because of missing terminfo search the kitty FAQ for sudo.

**Comment by page-down** [comment]
> ... I can select different colorschemes, but they all turn out black and white.

It appears that vim does not know if the current terminal supports colors.

Did you try the manual installation steps described in the FAQ for `terminfo`? Search for `sudo`.

https://sw.kovidgoyal.net/kitty/faq/

**Comment by waynongithub** [comment]
Thanks for replying!

I've taken these steps that I found in the FAQ:

1. check TERM and TERMINFO
```
$ printenv | grep TERM
   COLORTERM=truecolor
   TERMINFO=/media/bconf/settings/kitty.app/lib/kitty/terminfo
   TERM=xterm-kitty
```
2. Verify that x/xterm-kitty is indeed in the directory that TERMINFO is pointing to:
```
/media/bconf/settings/kitty.app/lib/kitty/terminfo/x $ ls
xterm-kitty
```
3. I'm on Linux, so I did not add ~/.terminfo as the instructions in the FAQ state that this step is only for MACOS

4. In `sudo visudo`, I added 
`Defaults env_keep += "TERM TERMINFO"`


I rebooted and started kitty with `kitty --config NONE` but `sudo vim` still renders in black and white. 
In desperation I did try the steps for MACOS:
```
mkdir -p ~/.terminfo/{78,x}
ln -snf ../x/xterm-kitty ~/.terminfo/78/xterm-kitty
tic -x -o ~/.terminfo "$KITTY_INSTALLATION_DIR/terminfo/kitty.terminfo"
```
And now it works. So the missing step was that adding ~/.terminfo is not just for MACOS, but apparently for some linux systems as well.

Thanks again!
